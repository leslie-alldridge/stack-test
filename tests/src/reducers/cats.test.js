import cats from "../../../src/reducers/cats";
import {
  loading,
  errorMessage,
  receiveCats
} from "../../../src/actions/getAll";

describe("cat reducer", () => {
  const catData = [
    {
      id: 1,
      name: "tester",
      location: "nz",
      age: 32
    }
  ];
  it("returns the initial state", () => {
    expect(cats(undefined, {})).toMatchSnapshot();
  });

  it("handles the loading action", () => {
    expect(cats({}, loading())).toMatchSnapshot();
  });

  it("handles the error action", () => {
    expect(cats({}, errorMessage("error"))).toMatchSnapshot();
  });

  it("handles the loading action", () => {
    expect(cats({}, receiveCats(catData))).toMatchSnapshot();
  });
});
