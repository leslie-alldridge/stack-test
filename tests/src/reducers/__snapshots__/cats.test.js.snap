// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cat reducer handles the error action 1`] = `
Object {
  "err": "error",
  "err2": undefined,
  "loading": false,
}
`;

exports[`cat reducer handles the loading action 1`] = `
Object {
  "err": null,
  "err2": null,
  "loading": true,
}
`;

exports[`cat reducer handles the loading action 2`] = `
Object {
  "catData": Array [
    Object {
      "age": 32,
      "id": 1,
      "location": "nz",
      "name": "tester",
    },
  ],
  "err": null,
  "err2": null,
  "loading": false,
}
`;

exports[`cat reducer returns the initial state 1`] = `
Object {
  "catData": Array [],
  "loading": false,
}
`;
